pkg/distributor/distributor_test.go

```diff
@@ -277,10 +316,50 @@ func TestDistributor_Push(t *testing.T) {
 					return testutil.GatherAndCompare(regs[0], strings.NewReader(tc.expectedMetrics), tc.metricNames...)
 				})
 			}
+
+			const eventName = "Distributor.prePushValidationMiddleware[deduplicated samples/histograms with conflicting timestamps from write request]"
+			event := findSpanEvent(t, spanRecorder, span, spanName, eventName)
+			if !tc.expectDedupTraceEvent {
+				require.Emptyf(t, event.Name, "Expected no sample deduplication trace events")
+			} else {
+				verifySpanEvent(t, event, eventName, []attribute.KeyValue{
+					attribute.String("metric", "test_metric"),
+					attribute.Int64("count", 1),
+				})
+			}
 		})
 	}
 }
 
+func verifySpanEvent(t *testing.T, event trace.Event, name string, attributes []attribute.KeyValue) {
+	t.Helper()
+
+	assert.Equal(t, name, event.Name)
+	assert.Equal(t, attributes, event.Attributes)
+}
+
+func findSpanEvent(t *testing.T, spanRecorder *tracetest.SpanRecorder, span oteltrace.Span, spanName, eventName string) trace.Event {
+	t.Helper()
+
+	span.End()
```

Comment:
Why are we passing the `span` to this method? Why is this method ending the span? Seems to be totally unrelated.

Reply:
Good point, I will fix it.

Reply:
Done. PTAL.